runners/direct-java/src/test/java/org/apache/beam/runners/direct/DirectRunnerApiSurfaceTest.java

```diff
@@ -43,9 +48,17 @@ public void testDirectRunnerApiSurface() throws Exception {
     final ClassLoader thisClassLoader = getClass().getClassLoader();
     ApiSurface apiSurface =
         ApiSurface.ofPackage(thisPackage, thisClassLoader)
+            // Do not include dependencies that are required based on the known exposures. This
+            // could alternatively prune everything exposed by the public parts of the Core SDK
+            .pruningClass(Pipeline.class)
```

Comment:
This seems OK, though `org.apache.beam.sdk` is already allowed so it should be made automatic. Perhaps this is part of [BEAM-2231](https://issues.apache.org/jira/browse/BEAM-2231), making is lazy. Whenever you arrive at a class that is allowed, it should be presumed anything reachable from there is also allowed. Of course, this would make us miss when one of our dependencies mistakenly exposes Guava, for example.